src/ensembl/src/shared/species-tab/SpeciesTab.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
+import styles from './SpeciesTab.scss';
+
+import { CommittedItem } from 'src/content/app/species-selector/types/species-search';
+
+type Props = {
+  species: CommittedItem;
+  isActive: boolean;
+  onActivate: (genomeId: string) => void;
+};
+
+const SpeciesTab = (props: Props) => {
+  const [isHovering, setIsHovering] = useState(false);
+
+  const {
+    genome_id,
+    common_name,
+    scientific_name,
+    assembly_name
+  } = props.species;
+
+  const toggleHoverState = (newHoverState: boolean) => {
+    if (newHoverState !== isHovering) {
+      setIsHovering(newHoverState);
+    }
```

Comment:
What if `newHoverState` is equal to `isHovering`? Wouldn't you wanna toggle the hover state?

Reply:
From lines 45-50:

```
    <div
      className={className}
      onMouseEnter={() => toggleHoverState(true)}
      onMouseLeave={() => toggleHoverState(false)}
      onClick={handleClick}
    >
```

I want to switch isHovering to `true` only on onMouseEnter. And I want to switch it to `false` only on onMouseLeave. If it is already `true` when onMouseEnter is fired or `false` when onMouseLeave is fired (I can't imagine how this can happen, but just in case it can), then I am ignoring the call to the function.

Most likely, I could have done just with:

```
      onMouseEnter={toggleHoverState}
      onMouseLeave={toggleHoverState}
```

and then in the `toggleHoverState` function always switch the state to its opposite, but it feels a little bit less robust than the current version.

Reply:
Ah sorry, I misunderstood the functionality. This makes sense.